internal/controllers/topology/cluster/desired_state.go

```diff
@@ -897,19 +908,29 @@ func templateToTemplate(in templateToInput) *unstructured.Unstructured {
 	// Enforce the topology labels into the provided label set.
 	// NOTE: The cluster label is added at creation time so this object could be read by the ClusterTopology
 	// controller immediately after creation, even before other controllers are going to add the label (if missing).
-	labels := template.GetLabels()
-	if labels == nil {
-		labels = map[string]string{}
+	labels := map[string]string{}
+	for k, v := range in.labels {
+		labels[k] = v
+	}
+	if template.GetLabels() != nil {
+		for k, v := range template.GetLabels() {
+			labels[k] = v
+		}
 	}
```

Comment:
Included this only because `labels` and `annotations` are now supported in the input and did not want to be inconsistent in the behavior.

The labels and annotations are not used in any of the existing `templateToTemplate` usages.

Reply:
Makes sense!

Reply:
We could also use different input structs if we don't want to introduce "dead" code

(I'm fine either way)